lmdeploy/pytorch/backends/selector.py

```diff
@@ -31,6 +31,12 @@ def get_backend(backend_type: str = None):
         return _get_backend()
     else:
         device_ctx = DeviceContext(backend_type)
-        device_mgr = get_device_manager()
+        device_mgr = get_device_manager(device_ctx)
```

Comment:
the `with` conext would switch the device, there is no need to init the device manager with the context.

Reply:
The default device_type is cuda and the `with context` not set backend_type to DeviceManager. When get_backend() is called without passing the backend_type parameter, it returns 'cuda' instead of the expected 'ascned'.

Reply:
we assume `get_backend()` should alway be called inside the `with` context.

Reply:
ok, i already remove this